integration/presto/src/main/prestodb/org/apache/carbondata/presto/CarbondataSplitManager.java

```diff
@@ -109,8 +109,14 @@ public ConnectorSplitSource getSplits(ConnectorTransactionHandle transactionHand
     if (!table.getStorage().getStorageFormat().getInputFormat().contains("carbon")) {
       return super.getSplits(transactionHandle, session, layoutHandle, splitSchedulingStrategy);
     }
-    String location = table.getStorage().getLocation();
-
+    String location;
```

Comment:
assign with table.getStorage().getSerdeParameters().get("tablePath") first and then update it if null

Reply:
done